src/Bridges/Psr/Cache.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/**
+ * This file is part of the Tracy (https://tracy.nette.org)
+ * Copyright (c) 2004 David Grudl (https://davidgrudl.com)
+ */
+
+declare(strict_types=1);
+
+namespace Nette\Bridges\Psr;
+
+use DateInterval;
+use Generator;
+use Nette;
+use Psr;
+
+class Cache implements Psr\SimpleCache\CacheInterface
+{
+	public function __construct(private Nette\Caching\Storage $storage)
+	{
+	}
+
+
+	public function get(string $key, mixed $default = null): mixed
+	{
+		return $this->storage->read($key) ?? $default;
+	}
+
+
+	public function set(string $key, mixed $value, null|int|DateInterval $ttl = null): bool
+	{
+		$dependencies = [];
+		if ($ttl !== null) {
+			$dependencies[Nette\Caching\Cache::Expire] = self::ttlToSeconds($ttl);
+		}
+
+		$this->storage->write($key, $value, $dependencies);
+
+		return true;
+	}
+
+
+	public function delete(string $key): bool
+	{
+		$this->storage->remove($key);
+
+		return true;
+	}
+
+
+	public function clear(): bool
+	{
+		$this->storage->clean([Nette\Caching\Cache::All => true]);
+
+		return true;
+	}
+
+
+	/**
+	 * @return Generator<string, mixed>
+	 */
+	public function getMultiple(iterable $keys, mixed $default = null): iterable
+	{
+		foreach ($keys as $name) {
+			yield $name => $this->get($name, $default);
+		}
+	}
+
+
+	/**
+	 * @param iterable<string|int|float, mixed> $values
```

Comment:
Why allow `float` key?

Reply:
You have right, this is deprecated.

**E_DEPRECATED: Implicit conversion from float 5.2 to int loses precision**